test/EFCore.PG.FunctionalTests/Scaffolding/NpgsqlDatabaseModelFactoryTest.cs

```diff
@@ -1413,6 +1413,66 @@ public void Index_operators()
                 },
                 @"DROP TABLE ""IndexOperators""");
 
+        [Fact]
+        public void Index_collation()
+            => Test(@"
+CREATE TABLE ""IndexCollation"" (a text, b text);
+CREATE INDEX ix_with ON ""IndexCollation"" (a, b COLLATE ""POSIX"");
+CREATE INDEX ix_without ON ""IndexCollation"" (a, b);",
+                Enumerable.Empty<string>(),
+                Enumerable.Empty<string>(),
+                dbModel =>
+                {
+                    var table = dbModel.Tables.Single();
+
+                    var indexWith = table.Indexes.Single(i => i.Name == "ix_with");
+                    Assert.Equal(new[] { null, "POSIX" }, indexWith.FindAnnotation(NpgsqlAnnotationNames.IndexCollation).Value);
+
+                    var indexWithout = table.Indexes.Single(i => i.Name == "ix_without");
+                    Assert.Null(indexWithout.FindAnnotation(NpgsqlAnnotationNames.IndexCollation));
+                },
+                @"DROP TABLE ""IndexCollation""");
+
+        [Fact]
+        public void Index_sort_order()
```

Comment:
As I wrote before, I think this can be easily parameterized on the index method (for btree, gist...) as a way of confirming that our way of interpreting the bits in `indoption` is indeed universal.

Reply:
Like this; 558e79b?